### ISSUE TITLE
`Expected-Version` does not behave as one would expect

### ISSUE THREAD

**Issue by D4nte** [comit-network/comit-rs#1418]
the header only works based on a released binary.

However, if a user (ie, me):
- freshly release 0.10.0
- Install btsieve and cnd (0.10.0) bin
- Upgrade btsieve API (breaking change) and cnd uses of it on master
- Install btsieve bin

Then the `Expected-Version` header will expect 0.10.0, so it would think that my installed cnd and btsieve are compatible, when they are not.

Instead, it should be based on current git tag and if not available, git commit.

tl;dr: `Expected-Version` header check should be based on git references, not `CARGO_PKG_VERSION`.

**Comment by thomaseizinger** [comment]
> the header only works based on a released binary.

I think that was the intention.
The `Expected-Version` header is a protection for users that use released versions, because it _may_ happen that they grab binaries of different releases.
Developers always compile anyway. I don't think we need to account for a scenario where you compile only one part of the software and use a released binary of the other part.

Since btsieve is going away very soon (#1408), I don't think it is worth changing the behavior here :)

**Comment by bonomat** [comment]
Closing this as this will be obsolete with #1408
